cmd/csi-attacher/main.go

```diff
@@ -279,6 +280,18 @@ func main() {
 		*reconcileSync,
 	)
 
+	// handle SIGTERM and SIGINT by cancelling the context.
+	shutdownCtx, cancel := context.WithCancel(ctx)
+	shutdownHandler := server.SetupSignalHandler()
+	ctx, terminate := context.WithCancel(shutdownCtx)
```

Comment:
That ctx handling looks weird. We wrap existing `ctx` with `WithCancel` and then wrap it again with - `WithCancel`. Is that expected?

Reply:
I cannot think of an answer from the top of my head, but I think that is a remainder I forgot to delete, I will double check and edit the code or give an explanation why we should use it.